cmd/admin-handlers-users.go

```diff
@@ -522,11 +522,16 @@ func (a adminAPIHandlers) AddServiceAccount(w http.ResponseWriter, r *http.Reque
 		targetGroups []string
 	)
 
+	// If the request did not set a TargetUser, the service account is
+	// created for the request sender.
 	targetUser = createReq.TargetUser
+	if targetUser == "" {
+		targetUser = cred.AccessKey
```

Comment:
This is not the targetUser for the service account @donatello it is cred.ParentUser, I hope this gets replaced subsequently later on in the code.